src/art.rs

```diff
@@ -746,53 +746,47 @@ impl<P: KeyTrait, V: Clone> Node<P, V> {
         cur_node_prefix: &P,
         key: &'a P,
         depth: usize,
-    ) -> (&'a [u8], P, P, bool, usize) {
+    ) -> (
+        &'a [u8], // remaining_key_suffix
+        P,        // remaining_node_suffix
+        P,        // shared_prefix
+        bool,     // is_prefix_match
+        usize,    // shared_prefix_length
+    ) {
         // Obtain the current node's prefix and its length.
         let cur_node_prefix_len = cur_node_prefix.len();
 
         // Determine the prefix of the key after the current depth.
-        let key_prefix = key.prefix_after(depth);
+        let remaining_key_suffix = key.prefix_after(depth);
+
         // Find the longest common prefix between the current node's prefix and the key's prefix.
-        let longest_common_prefix = cur_node_prefix.longest_common_prefix(key_prefix);
+        let shared_prefix_length = cur_node_prefix.longest_common_prefix(remaining_key_suffix);
 
         // Create a new key that represents the remaining part of the current node's prefix after the common prefix.
-        let mut new_key = cur_node_prefix.prefix_after(longest_common_prefix);
-        if new_key.is_empty() {
+        let mut remaining_node_suffix = cur_node_prefix.prefix_after(shared_prefix_length);
+        if remaining_node_suffix.is_empty() {
             // In the case where the current node's prefix is shorter than the key,
             // the suffix comes from the key_prefix. For example, cur_node.prefix="key"
             // and key="keyboard", then new_key should be "board".
```

Comment:
`new_key` should be changed to `remaining_node_suffix `.

Reply:
Done!